### DISCUSSION TITLE
求助“无法识别固件版本，请更新硬盘数据库”

### DISCUSSION THREAD

**Discussion by ddw0822** [007revad/Synology_HDD_db#298] [Q&A]
型号：DS923+
系统：DSM 7.2.1-69057 Update 5
问题描述：存储管理  →  HDD/SSD  →  固件版本显示：“无法识别固件版本，请更新硬盘数据库”
顺便请教一下，syno_hdd_vendor_ids.txt这个文件怎么修改呢？
![微信截图_20240525125403](https://github.com/007revad/Synology_HDD_db/assets/77032946/5e5ba60d-3884-40b2-bd34-2e01e1e70d2f)

**Comment by ddw0822** [comment]
补充执行日志：
HDD/SSD models found: 1
HUH721212ALE600,LEBDT6B0

M.2 drive models found: 1
Lexar SSD ARES 2TB,11296

No M.2 PCIe cards found

No Expansion Units found

Added HUH721212ALE600 to ds923+_host_v7.db
Edited unverified drives in ds923+_host_v7.db
Added Lexar SSD ARES 2TB to ds923+_host_v7.db

Support disk compatibility already disabled.

Disabled support memory compatibility.

Max memory is set to 32 GB.

NVMe support already enabled.

M.2 volume support already enabled.

Drive db auto updates already enabled.

DSM successfully checked disk compatibility.

You may need to reboot the Synology to see the changes.

**Comment by 007revad** [comment]
You only need to edit the syno_hdd_vendor_ids.txt file if the script warns you about a missing vendor id. If that happens the warning will show you the vendor id like:
```
WARNING No vendor found for vid 0x05dc
You can add 0x05dc and your drive's vendor to syno_hdd_vendor_ids.txt
```
What does the following commands return?
```
nvme id-ctrl /dev/nvme0 | grep -E ^vid
nvme id-ctrl /dev/nvme0 | grep -E ^mn
nvme id-ctrl /dev/nvme0 | grep -E '^fr '
grep -i lexar /usr/syno/etc.defaults/pci_vendor_ids.conf
```

**Reply by ddw0822** [reply]
感谢您的回复，执行结果如下：
root@dw_Synology:~# nvme id-ctrl /dev/nvme0 | grep -E ^vid
vid     : 0x1d97
root@dw_Synology:~# nvme id-ctrl /dev/nvme0 | grep -E ^mn
mn      : Lexar SSD ARES 2TB                      
root@dw_Synology:~# nvme id-ctrl /dev/nvme0 | grep -E '^fr '
fr      : 11296   
root@dw_Synology:~# grep -i lexar /usr/syno/etc.defaults/pci_vendor_ids.conf
0x1d97="SPCC/Lexar"
![001](https://github.com/007revad/Synology_HDD_db/assets/77032946/24bbf724-a66a-4761-a019-bd455f4fc03a)

**Reply by ddw0822** [reply]
我已经将 0x1d97="SPCC/Lexar" 添加到 syno_hdd_vendor_ids.txt 文件，然后重新执行了脚本 syno_hdd_db.sh，重新启动了系统。但是仍旧显示：“无法识别固件版本，请更新硬盘数据库”。
请您继续帮忙指点下。
![002](https://github.com/007revad/Synology_HDD_db/assets/77032946/0690f621-781e-4d81-aaa9-e7c7686b2335)
![003](https://github.com/007revad/Synology_HDD_db/assets/77032946/f233d7e5-63f1-4910-b2c7-c7fe9bc3490d)

**Reply by 007revad** [reply]
Run the script with the -n option to prevent DSM updating the drive database.
```
./syno_hdd_db.sh -n
```

**Reply by ddw0822** [reply]
按照您的方法执行了 ./syno_hdd_db.sh -n，并重新启动了系统 reboot，问题依旧存在。
![004](https://github.com/007revad/Synology_HDD_db/assets/77032946/5a714ac7-3127-495e-94ae-ff8887aa9b30)
![005](https://github.com/007revad/Synology_HDD_db/assets/77032946/5e526ab2-858e-4622-a024-4b6125a1bfd0)

**Reply by 007revad** [reply]
What does this command return?
```
jq -r --arg drive_model 'Lexar SSD ARES 2TB' '.disk_compatbility_info["Lexar SSD ARES 2TB"]' /var/lib/disk-compatibility/ds923+_host_v7.db
```

**Reply by ddw0822** [reply]
结果如下：
![image](https://github.com/007revad/Synology_HDD_db/assets/77032946/5e8493a7-9785-45e0-a0e0-47afa203a688)

**Reply by 007revad** [reply]
Everything is okay there.

Can you run these commands:
```
sudo sed -i "/drive_db_test_url=*/d" "/etc.defaults/synoinfo.conf"
sudo sed -i "/drive_db_test_url=*/d" /etc/synoinfo.conf
```
Then open storage manager and:
1. Click on HDD/SSD.
2. Click on the Settings button.
3. Click on Advanced.
4. Under "Drive Database" click on the Update Now button.
5. Wait for it to finish updating.
6. Execute `./syno_hdd_db.sh -n`
7. Reboot.

**Reply by 007revad** [reply]
You could also try:
```
sudo synosetkeyvalue /usr/syno/etc.defaults/pci_vendor_ids.conf 0x1d97 Lexar
sudo synosetkeyvalue /usr/syno/etc/pci_vendor_ids.conf 0x1d97 Lexar
```
Then close and open storage manager.
